cpp/include/raft/sparse/cusparse_wrappers.h

```diff
@@ -26,6 +26,26 @@
   case err:                       \
     return #err;
 
+//Notes:
+//(1.) CUDA_VER_SELECT aggregates all the CUDA version selection logic;
+//(2.) to enforce a lower version,
+//
+//`#define CUDA_ENFORCE_LOWER
+// #include <raft/sparse/cusparse_wrappers.h>`
+//
+// (i.e., before including this header)
+//
+#define CUDA_VER_SELECT         \
+  (__CUDACC_VER_MAJOR__ > 10 or \
```

Comment:
What about using CUSPARSE_VER_MAJOR instead? It would be nice not to require users to only include this in .cu, since .cu files can be slower to compile.

Reply:
It looks like `CUSPARSE_VER_*` are not in sync w/ `__CUDACC_VER_*__`. For a CUDA 11.0 build I get:
```pragma message: CUSPARSE_VER_MAJOR=11
                                              ^
#pragma message: CUSPARSE_VER_MINOR=1```

So, the minor is not the same. Which then begs the question if the major might also get out of sync.

Reply:
Therefore, I've adopted the `CUDART_VERSION` approach.